zipkin-collector/kafka/src/test/java/zipkin2/collector/kafka/ITKafkaCollector.java

```diff
@@ -68,17 +66,18 @@
   };
   KafkaProducer<byte[], byte[]> producer;
 
-  @Before public void setup() {
-    final Properties config = new Properties();
-    config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.bootstrapServers());
+  @BeforeEach void setup() {
+    metrics.clear();
+    Properties config = new Properties();
+    config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.bootstrapServer());
     producer = new KafkaProducer<>(config, new ByteArraySerializer(), new ByteArraySerializer());
   }
 
-  @After public void teardown() {
-    producer.close();
+  @AfterEach void tearDown() {
+    if (producer != null) producer.close();
```

Comment:
Doesn't hurt but just curious, is producer ever null?

Reply:
I don't know if aftereach runs when beforeeach fails, if so, yeah it could be null